README.md

```diff
@@ -11,38 +11,49 @@ python3 -m pip install -r requirements-test.txt
 tox
 ```
 
-## Local test by running the migration tool
+## Integration test
+
+pipeline-migration-tool relies on task bundles are annotated and migrations are attached properly.
+This integration test sets up a testing environment, inside which tasks are built and pushed by the
+`build-and-push.sh` script.
+
+Prerequisite:
+
+- A local clone of [konflux-ci/build-definitions](https://github.com/konflux-ci/build-definitions) 
+  and checkout to `main` branch.
+- Create public image repositories `task-clone` and `task-lint` under specified `QUAY_NAMESPACE`.
+- Log into Quay.io in order to make `tkn-bundle-push` work.
+
+For different test scenarios, the task bundles can be customized by setting `RECIPE` environment
+variable for `hack/integration-test/setup.sh`. A recipe is a multi-lines text and each line consists
+of three fields in order, task name, task version, and marker indicating whether the task bundle
+build should have a migration. For example:
+
+```bash
+RECIPE="
+clone 0.3 M
+lint 0.3 -
+"
+```
+
+Example steps to run the test:
 
 ```bash
 python3 -m venv venv
 source ./venv/bin/activate
 python3 -m pip install -r requirements.txt
 python3 -m pip install -e .
 
-# Log into a registry
-podman login quay.io
-
-# Push sample task bundles to your image repositories
-# For example, the following set will result in bundles like quay.io/account_name/task-clone
-export IMAGE_NS=quay.io/account_name
-
-# Enable local test in order to work with images from arbitrary image organization.
-export PMT_LOCAL_TEST=on
-
-# Specify an alternative registry authentication file
-export REGISTRY_AUTH_JSON=path/to/auth.json
-
-# build and push sample task bundles task-clone and task-tests.
-bash ./hack/local-test/run.sh build-and-push
+git checkout -b <test branch>  # setup.sh commits changes to the repo
 
-# Make sure the image repositories are public, especially if you did not create them before running the above command.
-# Sample Renovate upgrades data is written into file /tmp/pmt-upgrades-data.txt
+BUILD_DEFS_REPO=<path to build-definitions> \
+QUAY_NAMESPACE=<quay namespace passed to build-definitions/hack/build-and-push.sh> \
+./hack/integration-test/setup.sh
 
-# Run the migration tool
-pipeline-migration-tool -u "$(cat /tmp/pmt-upgrades-data.txt)"
+cd ./hack/integration-test/app
+PMT_LOCAL_TEST=1 pipeline-migration-tool -u "$(cat /tmp/pmt-test-upgrades.txt)"
```

Comment:
I got this error
```
pipeline_migration.migrate.IncorrectMigrationAttachment: 3 referrers containing migration script are listed. However, there should be one per task bundle.
```
Not sure what I did wrong yet. It might be that I run `setup.sh` multiple times, checking

Reply:
How did you run the setup.sh? I added a comment that tells to empty the image repositories, which should have been added when open.

I ran the setup.sh twice without cleaning up the image repo, the issue didn't happen to me.

Did you use a recipe like:

```
...
clone 0.2 M
lint 0.2 M
...
```

In this case, the migration will be attached to clone bundle twice.

Reply:
> How did you run the setup.sh?

Just this :confused: 

```bash
BUILD_DEFS_REPO=$(realpath ../build-definitions) \
QUAY_NAMESPACE=acmiel-test/tekton-catalog \
bash hack/integration-test/setup.sh
```

Reply:
Ah, I think what happened is that one of the repositories was private when I first ran setup.sh. I haven't been able to reproduce it since.